### PR TITLE
kubectl describe <kind> work with no arguments

### PR DESCRIPTION
**Pull request by feihujiang**
ref #10305

### CODE REVIEW

**Comment by feihujiang** [comment]
@bgrant0607 @jlowdermilk Could you please review this PR?

**Comment by ncdc** [comment]
cc @jwforres @liggitt

**Comment by jlowdermilk** [comment]
Thanks @feihujiang. Can you add a test case for `kubectl describe <kind>` with no arguments in `hack/test-cmd.sh`?

**Comment by feihujiang** [comment]
@jlowdermilk OK, I will add it.

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -236,6 +236,7 @@ runTests() {
   kube::test::get_object_assert 'pods/valid-pod' "{{$id_field}}" 'valid-pod'
   # Describe command should print detailed information
   kube::test::describe_object_assert pods 'valid-pod' "Name:" "Image(s):" "Node:" "Labels:" "Status:" "Replication Controllers"
+  kubectl describe "${kube_flags[@]}" pods
```
Comment (jlowdermilk) [inline hack/test-cmd.sh]: This checks that command runs and exits 0, but not output. I sent you a short pr to improve coverage a bit.

**Comment by jlowdermilk** [comment]
Needs squash and rebase.

**Comment by feihujiang** [comment]
@jlowdermilk Yes, need squash. But when I rebased onto master, I found something wrong, like cmd.Flags().StringSliceP undefined, cmd.Flags().SetAnnotation undefined etc. I don't know what's wrong with them. I guess I should update golang version?  My Golang version is 1.4.0.

**Comment by feihujiang** [comment]
@jlowdermilk I squashed and rebased onto master, please review.

**Comment by jlowdermilk** [comment]
LGTM

**Comment by satnam6502** [comment]
@k8s-bot ok to test

**Comment by satnam6502** [comment]
Please can someone look into the Shippable failure. Thank you.

**Comment by jlowdermilk** [comment]
Shippable failure was flake in a mesos test (unrelated to this pr). Needs rebase anyway, so won't bother kicking manually.

**Comment by feihujiang** [comment]
@satnam6502 I have rebased, and Shippable builds completed.  @jlowdermilk Thanks for looking into the Shippable failure.

**Pull request merged**